runtime/src/main/java/org/corfudb/runtime/clients/ManagementHandler.java

```diff
@@ -48,7 +51,11 @@ private static Object handleOrchestratorResponse(CorfuPayloadMsg<OrchestratorRes
     @ClientHandler(type = CorfuMsgType.HEARTBEAT_RESPONSE)
     private static Object handleHeartbeatResponse(CorfuPayloadMsg<byte[]> msg,
                                                   ChannelHandlerContext ctx, IClientRouter r) {
-        return msg.getPayload();
+        try {
+            return NodeView.parseFrom(msg.getPayload());
+        } catch (InvalidProtocolBufferException e) {
+            throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)